server/internal/usecase/repo/thread.go

```diff
@@ -0,0 +1,22 @@
+package repo
+
+import (
+	"context"
+	"errors"
+
+	"github.com/reearth/reearth-cms/server/pkg/id"
+	"github.com/reearth/reearth-cms/server/pkg/thread"
+)
+
+var (
+	ErrCommentNotFound error = errors.New("comment not found")
+)
+
+type Thread interface {
+	Save(context.Context, *thread.Thread) error
+	Filtered(filter WorkspaceFilter) Thread
+	FindByID(ctx context.Context, id id.ThreadID) (*thread.Thread, error)
+	AddComment(context.Context, *thread.Thread, *thread.Comment) error
+	UpdateComment(context.Context, *thread.Thread, *thread.Comment) error
+	DeleteComment(context.Context, *thread.Thread, id.CommentID) error
```

Comment:
```suggestion
```

On second thought, there is no problem without these methods; Thread is an aggregate and a root entity. Comments are internal entities belonging to Thread. In principle, it is safer to add, update, and delete comments through the methods of Thread because the business logic is consolidated into one.

- add: `threadRepo.FindByID(ctx, id)` -> `thread.AddComment(comment)` -> `threadRepo.Save(thread)`
- update: `threadRepo.FindByID(ctx, id)` -> `thread.UpdateComment(comment)` -> `threadRepo.Save(thread)`
- delete: `threadRepo.FindByID(ctx, id)` -> `thread.DeleteComment(comment)` -> `threadRepo.Save(thread)`

Reply:
So, for thread it should be like this:
```
	Add(context.Context, *thread.Thread, *thread.Comment) error
	Update(context.Context, *thread.Thread, *thread.Comment) error
	Delete(context.Context, *thread.Thread, id.CommentID) error
```
and then internally inside thread add these methods
```
	AddComment(c *thread.Comment)
	UpdateComment(c *thread.Comment)
	DeleteComment(cid id.CommentID)
```
Right?

Reply:
Done!